omero/developers/PythonBlitzGateway.txt

```diff
@@ -1,19 +1,69 @@
+This page provides some background information on the OMERO Python client
+'gateway' (omero.gateway module).
+
+The Blitz Gateway is a Python client-side library that facilitates working
+with the OMERO API, handling connection to the server, loading of data objects
+and providing convenience methods to access the data. It was originally
+designed as part of the |OmeroWeb|, to provide connection and data
+retrieval services to various web clients. However, we
+encourage its use for all access to the OMERO Python API.
+
+
+
+Connection wrapper
+""""""""""""""""""
 
-Specifically, the API for the 'conn' connection wrapper created above is
-:pythondoc:`here <omero/omero.gateway.html#omero.gateway.BlitzGateway>`.
+The BlitzGateway class (see :pythondoc:`API of development code <omero/omero.gateway.html#omero.gateway._BlitzGateway>`)
+is a wrapper for the OMERO client and session objects. It provides
+various methods for connecting to the OMERO server, querying the status
+or context of the current connection and retrieving data objects from OMERO.
 
-When working with :javadoc:`OMERO model objects <slice2html/omero/model.html>`
-(omero.model.Image etc) the Gateway will wrap these objects in classes
-such as :pythondoc:`omero.gateway.ImageWrapper <omero/omero.gateway.html#omero.gateway.ImageWrapper>`
-to handle object loading and hierarchy traversal. For example:
+
+Model object wrappers
+"""""""""""""""""""""
+
+OMERO model objects, e.g. omero.model.Project, omero.model.Pixels etc.
+(see :javadoc:`full list <slice2html/omero/model.html>`)
+are code-generated and mapped to the OMERO database schema. They are
+language agnostic and their data is in the form of omero.rtypes as
+described in :ref:`about model objects <AdvancedClientDevelopment#Objects>`.
+
+To facilitate work in Python, particularly in web page templates, these
+Python model objects are wrapped in Blitz Object Wrappers. This hides
+the use of rtypes.
 
 ::
 
+    import omero
+    from omero.model import ProjectI
+    from omero.rtypes import rstring
+    p = ProjectI()
+    p.setName(rstring("Omero Model Project"))   # attributes are all rtypes
+    print p.getName().getValue()                # getValue() to unwrap the rtype
+    print p.name.val                            # short-hand
+
+    from omero.gateway import ProjectWrapper
+    project = ProjectWrapper(obj=p)             # wrap the model.object
+    project.setName("Project Wrapper")          # Don't need to use rtypes
+    print project.getName()
+    print project.name
+
+    print project._obj                  # access the wrapped object with ._obj
+
+These wrappers also have a reference to the BlitzGateway connection wrapper,
+so they can make calls to the server and load more data when needed (lazy
+loading).
+
+::
+
+    >>> from omero.gateway import BlitzGateway
+
+    >>> conn = BlitzGateway("username", "password", host="localhost", port=4064)
```

Comment:
we should certainly look at improving the various sections of the doc/scripts even the tests with context manager.
(test will be a good place to start since we will success/failure answer)